airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/source.py

```diff
@@ -0,0 +1,148 @@
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteCatalog is an object describing a list of all available streams in this source.
+            A stream is an AirbyteStream object that includes:
+            - its stream name (or table name in the case of Postgres)
+            - json_schema providing the specifications of expected schema for this stream (a list of columns described
+            by their names and types)
+        """
+        stream_name = DATASET_ITEMS_STREAM_NAME
+        json_schema = {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "type": "object",
+        }
+
+        return AirbyteCatalog(
+            streams=[AirbyteStream(name=stream_name, supported_sync_modes=[SyncMode.full_refresh], json_schema=json_schema)]
+        )
+
+    def read(
+        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
+    ) -> Generator[AirbyteMessage, None, None]:
+        """
+        Returns a generator of the AirbyteMessages generated by reading the source with the given configuration,
+        catalog, and state.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+            the properties of the spec.json file
+        :param catalog: The input catalog is a ConfiguredAirbyteCatalog which is almost the same as AirbyteCatalog
+            returned by discover(), but
+        in addition, it's been configured in the UI! For each particular stream and field, there may have been provided
+        with extra modifications such as: filtering streams and/or columns out, renaming some entities, etc
+        :param state: When a Airbyte reads data from a source, it might need to keep a checkpoint cursor to resume
+            replication in the future from that saved checkpoint.
+            This is the object that is provided with state from previous runs and avoid replicating the entire set of
+            data everytime.
+
+        :return: A generator that produces a stream of AirbyteRecordMessage contained in AirbyteMessage object.
+        """
+        logger.info("Reading data from Apify dataset")
+
+        dataset_id = config["datasetId"]
+        clean = config.get("clean", False)
+
+        client = ApifyClient()
+        dataset_client = client.dataset(dataset_id)
+
+        # Get total number of items in dataset. This will be used in pagination
+        dataset = dataset_client.get()
+        num_items = dataset["itemCount"]
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
is it worth putting a `max_workers=` param in here? No is a valid answer

Reply:
I don't think so. This way it default to 5 * num cores which sounds reasonable to me